engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -4,99 +4,92 @@
+import org.terasology.gestalt.module.ModuleRegistry;
+import org.terasology.gestalt.module.TableModuleRegistry;
+import org.terasology.gestalt.module.dependencyresolution.DependencyInfo;
+import org.terasology.gestalt.module.dependencyresolution.DependencyResolver;
+import org.terasology.gestalt.module.dependencyresolution.ResolutionResult;
+import org.terasology.gestalt.module.sandbox.APIScanner;
+import org.terasology.gestalt.module.sandbox.ModuleSecurityManager;
+import org.terasology.gestalt.module.sandbox.ModuleSecurityPolicy;
+import org.terasology.gestalt.module.sandbox.PermissionProvider;
+import org.terasology.gestalt.module.sandbox.PermissionProviderFactory;
+import org.terasology.gestalt.module.sandbox.StandardPermissionProviderFactory;
+import org.terasology.gestalt.module.sandbox.WarnOnlyProviderFactory;
+import org.terasology.gestalt.naming.Name;
+
+import java.io.File;
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.Reader;
 import java.lang.reflect.ReflectPermission;
-import java.net.URISyntaxException;
-import java.net.URL;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.security.Policy;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.Enumeration;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static com.google.common.base.Verify.verifyNotNull;
-
 public class ModuleManager {
-    /** Set this environment variable to "true" to load all modules in the classpath by default. */
-    public final static String LOAD_CLASSPATH_MODULES_ENV = "TERASOLOGY_LOAD_CLASSPATH_MODULES";
+    /** Set this property to "true" to allow modules on the classpath. */
     public final static String LOAD_CLASSPATH_MODULES_PROPERTY = "org.terasology.load_classpath_modules";
 
     private static final Logger logger = LoggerFactory.getLogger(ModuleManager.class);
     private final StandardPermissionProviderFactory permissionProviderFactory = new StandardPermissionProviderFactory();
-    private final PermissionProviderFactory wrappingPermissionProviderFactory = new WarnOnlyProviderFactory(permissionProviderFactory);
+    private final PermissionProviderFactory wrappingPermissionProviderFactory =
+            new WarnOnlyProviderFactory(permissionProviderFactory);
 
     private final ModuleRegistry registry;
     private ModuleEnvironment environment;
     private final ModuleMetadataJsonAdapter metadataReader;
+    private final ModuleFactory moduleFactory;
     private final ModuleInstallManager installManager;
-    private Module engineModule;
+    private final Module engineModule;
 
     public ModuleManager(String masterServerAddress) {
         this(masterServerAddress, Collections.emptyList());
     }
 
     public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspathsToAddToEngine) {
```

Comment:
What is the `masterServer`? Does the module manager connect to a server to fetch modules from there? 

I saw a test where you pass in the empty string `""` - I guess to denote that there is no master server. Why not use `null` to denote that? (It probably was like that before, just trying to understand this).

Reply:
As you guessed, this is something that is unchanged in this branch. It is a good question that I have not explored the answer to.

But yeah, one of the things the ModuleManager does is set up that `ModuleInstallManager`, which is supposed to handle installation of modules at run-time?